### PR TITLE
Document Service Traffic Policies

### PR DESCRIPTION
**Pull request by andrewsykim**
Signed-off-by: Andrew Sy Kim <kim.andrewsy@gmail.com>

Documents traffic policies that can be set for Services. This PR adds documentation for both `internalTrafficPolicy` and `externalTrafficPolicy`. `externalTrafficPolicy` has always existed and is not new, but I realized the services document didn't have any explicit mention of it so I added it here. `internalTrafficPolicy` is a new feature we promoted to beta in v1.22. This PR also documents the new ProxyTerminatingEndpoints feature (alpha) and it's relation to externalTrafficPolicy.

Related KEPS:
https://github.com/kubernetes/enhancements/issues/2086
https://github.com/kubernetes/enhancements/issues/1669

### CODE REVIEW

File: content/en/docs/concepts/services-networking/service.md [hunk 1]
```diff
@@ -374,6 +374,33 @@ The IP address that you choose must be a valid IPv4 or IPv6 address from within
...
+### External Traffic Policy
+
+You can set the `spec.externalTrafficPolicy` field to control how traffic from external sources is routed.
+Valid values are `Cluster` and `Local`. Set the field to `Cluster` to route external traffic to all ready endpoints
+and `Local` to only route to ready node-local endpoints. If the traffic policy is `Local` and there are are no node-local
+endpoints, traffic is dropped by kube-proxy.
+
+{{< note >}}
+{{< feature-state for_k8s_version="v1.22" state="alpha" >}}
+If the feature gate ProxyTerminatingEndpoints is enabled from kube-proxy, it will route traffic to terminating endpoints
```
Comment (andrewsykim) [inline content/en/docs/concepts/services-networking/service.md:388]: I struggled a bit with writing this section, the problem is quite nuanced and hard to articulate in a simple way. Definitely open to suggestions on how to improve the wording here

**Comment by andrewsykim** [comment]
/assign @thockin @robscott @maplain 

For technical review

**Comment by PI-Victor** [comment]
@andrewsykim please change the branch to `dev-1.22`

/assign @carlisia @ritpanjw 
@carlisia  @ritpanjw  since this covers both #1669 and #2086 in k/enhancements i'm assign this to both
/sig network

**Comment by PI-Victor** [comment]
@andrewsykim please change the branch to `dev-1.22` this PR is targeted at the wrong branch

**Comment by andrewsykim** [comment]
Sorry, let me do that now

**Comment by andrewsykim** [comment]
https://github.com/kubernetes/website/pull/29121

**Pull request closed without merging**